src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +515,102 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isInProc, bool isVerbose)
+        {
+            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isInProc)
+                {
+                    throw new CliException($"The runtime host value passed in, default, is not a valid host version for your project. For the default host runtime, the worker runtime must be set to {WorkerRuntime.dotnetIsolated}.");
```

Comment:
There's a mix of `HostRuntime` validation in `RunAsync` and this method. Can we consolidate all this (moving this to a dedicated method?